### DISCUSSION TITLE
Cannot cache component

### DISCUSSION THREAD

**Discussion by gitkwarteng** [django-commons/django-unicorn#305] [Q&A] [answered]
I noticed that component gets instantiated each time it processes a request. Then I found this error in the console.
"Cannot cache component because it is not picklable:". what could I be doing wrong and what do I do to solve it. Thanks.

**Comment by adamghill** [comment, marked as answer]
@mailantwi Hello! Unicorn caches components so it can skip instantiating them on every request and only update the fields that change. There should be a field in the error message that can give a clue why the component cannot be cached. Most Python primitives, Django models, etc should be pickleable, but if you can give me more information from the error message I might be able to figure out what is going on.

**Reply by gitkwarteng** [reply]
Thanks for feedback. the full error is: Cannot cache component '<class 'unicorn.components.receive_inventory_search.ReceiveInventorySearchView'>' because it is not picklable: <class 'TypeError'>: cannot pickle '_io.BufferedReader' object

**Reply by gitkwarteng** [reply]
In my component, I was adding an instance of a custom class for manipulating session data as a property of the component on mount. And I noticed that was what wasn't picklable. removing that instance solved the problem. How do I make the class picklable ... ?
